### PR TITLE
update pessimistic transaction

### PR DESCRIPTION
**Pull request by youjiali1995**
Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What is changed, added or deleted? (Required)

Adjust docs about pessimistic transactions to match v4.0.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by kissmydb** [comment]
@zhenjiaogao

**Comment by coocood** [comment]
LGTM

File: pessimistic-transaction.md [hunk 1]
```diff
@@ -7,39 +7,45 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
 
 # TiDB 悲观事务模型
 
-在 v3.0.8 之前，TiDB 默认使用的乐观事务模式会导致事务提交时因为冲突而失败。为了保证事务的成功率，需要修改应用程序，加上重试的逻辑。悲观事务模式可以避免这个问题，应用程序无需添加重试逻辑，就可以正常执行。
+为了使 TiDB 的使用方式更加贴近传统数据库，降低用户迁移的成本，TiDB 自 v3.0 版本开始在乐观事务模型的基础上支持了悲观事务模型。本文将介绍 TiDB 悲观事务的相关特性。
 
-## 悲观事务的使用方法
+> **注意：**
+>
+> 自 v3.0.8 开始，新创建的 TiDB 集群默认使用悲观事务模型。但如果从 3.0.7 版本及之前的创建的集群升级到 >= 3.0.8 的版本，不会改变默认事务模型，即**只有新创建的集群才会默认使用悲观事务模型**。
```
Comment (zhenjiaogao) [inline pessimistic-transaction.md:14]:
```suggestion
> 自 v3.0.8 开始，新创建的 TiDB 集群默认使用悲观事务模型。但如果从 v3.0.7 版本及之前创建的集群升级到 >= v3.0.8 的版本，则不会改变默认的事务模型，即**只有新创建的集群才会默认使用悲观事务模型**。
```

File: pessimistic-transaction.md [hunk 2]
```diff
@@ -7,39 +7,45 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
-- 执行 `set @@tidb_txn_mode = 'pessimistic';`，使这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。
+{{< copyable "sql" >}}
 
-- 执行 `set @@global.tidb_txn_mode = 'pessimistic';`，使之后整个集群所有新创建 session 执行的所有显示事务（即非 autocommit 的事务）都会进入悲观事务模式。
+```sql
+set @@global.tidb_txn_mode = 'pessimistic';
+```
 
-在配置了 `global.tidb_txn_mode` 为 `pessimistic` 之后，默认进入悲观事务模式，但是可以用以下三种方式使事务进入乐观事务模式：
+除此之外，还可以执行如下 SQL 显式开启悲观事务：
```
Comment (zhenjiaogao) [inline pessimistic-transaction.md:26]:
```suggestion
除此之外，还可以执行如下 SQL 显式开启乐观事务：
```

File: pessimistic-transaction.md [hunk 3]
```diff
@@ -74,12 +81,38 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
```
Comment (zhenjiaogao) [inline pessimistic-transaction.md:81]: line 72 行建议事务隔离级别的风格一致，如将 “因此有些业务会通过将隔离级别降低至 READ COMMITTED” 修改为 “因此有些业务会通过将隔离级别降低至 Read Committed”

File: pessimistic-transaction.md [hunk 4]
```diff
@@ -7,39 +7,45 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
 
 # TiDB 悲观事务模型
 
-在 v3.0.8 之前，TiDB 默认使用的乐观事务模式会导致事务提交时因为冲突而失败。为了保证事务的成功率，需要修改应用程序，加上重试的逻辑。悲观事务模式可以避免这个问题，应用程序无需添加重试逻辑，就可以正常执行。
+为了使 TiDB 的使用方式更加贴近传统数据库，降低用户迁移的成本，TiDB 自 v3.0 版本开始在乐观事务模型的基础上支持了悲观事务模型。本文将介绍 TiDB 悲观事务的相关特性。
```
Comment (lilin90) [inline pessimistic-transaction.md:10]:
```suggestion
为了使 TiDB 的使用方式更加贴近传统数据库，降低用户迁移数据的成本，TiDB 自 v3.0 版本开始在乐观事务模型的基础上支持了悲观事务模型。本文将介绍 TiDB 悲观事务的相关特性。
```

File: pessimistic-transaction.md [hunk 5]
```diff
@@ -7,39 +7,45 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
+> **注意：**
+>
+> 自 v3.0.8 开始，新创建的 TiDB 集群默认使用悲观事务模型。但如果从 v3.0.7 版本及之前创建的集群升级到 >= v3.0.8 的版本，则不会改变默认的事务模型，即**只有新创建的集群才会默认使用悲观事务模型**。
 
-进入悲观事务模式有以下三种方式:
+## 事务模式的修改方法
 
-- 执行 `BEGIN PESSIMISTIC;` 语句开启的事务，会进入悲观事务模式。
-可以通过写成注释的形式 `BEGIN /*!90000 PESSIMISTIC */;` 来兼容 MySQL 语法。
+TiDB 提供了 [tidb_txn_mode](/tidb-specific-system-variables.md#tidb_txn_mode) 系统变量用于设置事务模式，执行下面命令即可使整个集群所有新创建 session 执行的所有显示事务（即非 autocommit 的事务）进入悲观事务模式:
```
Comment (lilin90) [inline pessimistic-transaction.md:18]:
```suggestion
你可以使用 [`tidb_txn_mode`](/tidb-specific-system-variables.md#tidb_txn_mode) 系统变量设置事务模式。执行以下命令，即可使整个集群中所有新创建 session 执行的所有显示事务（即非 autocommit 的事务）进入悲观事务模式：
```

File: pessimistic-transaction.md [hunk 6]
```diff
@@ -7,39 +7,45 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
-- 执行 `set @@tidb_txn_mode = 'pessimistic';`，使这个 session 执行的所有显式事务（即非 autocommit 的事务）都会进入悲观事务模式。
+{{< copyable "sql" >}}
 
-- 执行 `set @@global.tidb_txn_mode = 'pessimistic';`，使之后整个集群所有新创建 session 执行的所有显示事务（即非 autocommit 的事务）都会进入悲观事务模式。
+```sql
+set @@global.tidb_txn_mode = 'pessimistic';
+```
 
-在配置了 `global.tidb_txn_mode` 为 `pessimistic` 之后，默认进入悲观事务模式，但是可以用以下三种方式使事务进入乐观事务模式：
+除此之外，还可以执行如下 SQL 显式开启悲观事务：
```
Comment (lilin90) [inline pessimistic-transaction.md:26]:
```suggestion
除此之外，还可以执行以下 SQL 语句显式地开启悲观事务：
```

File: pessimistic-transaction.md [hunk 7]
```diff
@@ -74,12 +81,38 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
+
+## 隔离级别
+
+TiDB 在悲观事务模式下支持了 2 种隔离级别：
+
+1. 默认使用与 MySQL 行为相同的[可重复读隔离级别 (Repeatable Read)](/transaction-isolation-levels.md#可重复读隔离级别-repeatable-read)。
+
+    > **注意：**
+    >
+    > 在这种隔离级别下 DML 会基于已提交的最新数据来执行，行为与 MySQL 相同，但与 TiDB 乐观事务不同，请参考 [与 MySQL 可重复读隔离级别的区别](/transaction-isolation-levels.md#与-mysql-可重复读隔离级别的区别)。
```
Comment (lilin90) [inline pessimistic-transaction.md:100]:
```suggestion
    > 在这种隔离级别下，DML 操作会基于已提交的最新数据来执行，行为与 MySQL 相同，但与 TiDB 乐观事务不同，请参考[与 MySQL 可重复读隔离级别的区别](/transaction-isolation-levels.md#与-mysql-可重复读隔离级别的区别)。
```

File: pessimistic-transaction.md [hunk 8]
```diff
@@ -74,12 +81,38 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
+
+2. 使用 [SET TRANSACTION](/sql-statements/sql-statement-set-transaction.md) 可设置为 [读已提交隔离级别 (Read Committed)](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
```
Comment (lilin90) [inline pessimistic-transaction.md:102]:
```suggestion
2. 使用 [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) 语句可将隔离级别设置为 [读已提交隔离级别 (Read Committed)](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
```

File: pessimistic-transaction.md [hunk 9]
```diff
@@ -74,12 +81,38 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
+
+## Pipelined 加锁流程
+
+加悲观锁需要向 TiKV 写入数据，要经过 raft 提交并 apply 后才能返回，相比于乐观事务，不可避免的会增加部分延迟。为了降低加锁的开销，TiKV 实现了 pipelined 加锁流程：当数据满足加锁要求时，TiKV 立刻通知 TiDB 执行后面的请求，并异步写入悲观锁，从而降低大部分延迟，显著提升悲观事务的性能。但有较低概率悲观锁异步写入失败，可能会导致悲观事务提交失败。
```
Comment (lilin90) [inline pessimistic-transaction.md:106]:
```suggestion
加悲观锁需要向 TiKV 写入数据，要经过 Raft 提交并 apply 后才能返回，相比于乐观事务，不可避免的会增加部分延迟。为了降低加锁的开销，TiKV 实现了 pipelined 加锁流程：当数据满足加锁要求时，TiKV 立刻通知 TiDB 执行后面的请求，并异步写入悲观锁，从而降低大部分延迟，显著提升悲观事务的性能。但有较低概率悲观锁异步写入失败，可能会导致悲观事务提交失败。
```

File: pessimistic-transaction.md [hunk 10]
```diff
@@ -74,12 +81,38 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
+
+![Pipelined pessimistic lock](/media/pessimistic-transaction-pipelining.png)
+
+该功能默认关闭，可修改 TiKV 配置启用:
```
Comment (lilin90) [inline pessimistic-transaction.md:110]:
```suggestion
该功能默认关闭，可修改 TiKV 配置启用：
```

File: pessimistic-transaction.md (hunk 4 above)
Comment (youjiali1995) [reply pessimistic-transaction.md:10]: 这里不是指迁移数据，应该是用户迁移到 TiDB。用“迁移业务”的话感觉也不太对。

Comment (youjiali1995) [reply pessimistic-transaction.md:10]: @lilin90 PTAL again, thanks!

**Comment by jackysp** [comment]
PTAL @lilin90

File: pessimistic-transaction.md [hunk 11]
```diff
@@ -74,12 +81,38 @@ aliases: ['/docs-cn/dev/reference/transactions/transaction-pessimistic/']
...
+
+TiDB 在悲观事务模式下支持了 2 种隔离级别：
+
+1. 默认使用与 MySQL 行为相同的[可重复读隔离级别 (Repeatable Read)](/transaction-isolation-levels.md#可重复读隔离级别-repeatable-read)。
+
+    > **注意：**
+    >
+    > 在这种隔离级别下，DML 操作会基于已提交的最新数据来执行，行为与 MySQL 相同，但与 TiDB 乐观事务不同，请参考[与 MySQL 可重复读隔离级别的区别](/transaction-isolation-levels.md#与-mysql-可重复读隔离级别的区别)。
+
+2. 使用 [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) 语句可将隔离级别设置为 [读已提交隔离级别 (Read Committed)](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
```
Comment (lilin90) [inline pessimistic-transaction.md:102]:
```suggestion
2. 使用 [`SET TRANSACTION`](/sql-statements/sql-statement-set-transaction.md) 语句可将隔离级别设置为[读已提交隔离级别 (Read Committed)](/transaction-isolation-levels.md#读已提交隔离级别-read-committed)。
```

**Pull request merged**